server/rest/src/main/java/org/infinispan/rest/operations/CacheOperations.java

```diff
@@ -330,8 +341,43 @@ public InfinispanResponse putValueToCache(InfinispanCacheAPIRequest request) thr
             Optional<Long> idle = request.getMaxIdleTimeSeconds();
             return putInCache(response, useAsync, cache, key, data, ttl, idle, oldData);
          }
-      } catch (CacheException cacheException) {
-         throw createResponseException(cacheException);
+      } catch (CacheException | IllegalStateException e) {
+         throw createResponseException(e);
+      }
+   }
+
+   public InfinispanSearchResponse search(InfinispanSearchRequest request, QueryRequest query) {
```

Comment:
How about putting this into a separate class? Cache operation would remain here and all Search operations would be put into `SearchOperations` class?

Reply:
It could be, as the amount of operations will grow very soon (admin, counter, scripts, etc)